metricflow/model/objects/data_source.py

```diff
@@ -51,6 +51,8 @@ class DataSource(HashableBaseModel, ParseableObject):
     """Describes a data source"""
 
     name: str
+    description: Optional[str]
+    owners: Optional[List[str]]
```

Comment:
Two questions:

1. Do we need owners at this point? If not, do we really want to add this right now?
2. This would be better as an `Optional[List[Owner]]` instead of an `Optional[List[str]]`, although I guess the bigger issue is with our config spec (where we require a List[str] in the YAML instead of a List[struct] or whatever it's called) than whatever it is we do here.

Reply:
1. I'm assuming we would want it, tho it was originally planned to be an internal field, @plypaul felt it would be better to move it to oss
2. I agree it would be better to change it to a struct. however, if we do, it would break existing configs as we would have to do something like this
```yaml
owners:
  - email: test@gmail.com
  - email: test2@gmail.com
```

rather than the current version,
```yaml
owners: 
  - test@gmail.com
  - test2@gmail.com
```
As for the  model parser, it parses everything that exists in the model object, so by adding the additional fields, it would automatically be parsing it. (Look at https://github.com/transform-data/metricflow/pull/67 for reference)

Reply:
Actually I got a bright idea and I'm experimenting with a way to make the array<string> -> array<struct> transition backwards compatible without adding a pile of internal technical debt, I just don't know if it works in our specific parsing/validation setup.

Separately, it appears there's some required/optional piece to this. I'll message you and if that works ok we can go as-is, but otherwise maybe we hold off on the owners field until we get that sorted out just so we can continue making progress.